app/modules/loan/endpoints_loan.py

```diff
@@ -828,6 +882,40 @@ async def return_loan(
         returned=True,
         returned_date=datetime.now(UTC),
     )
+    try:
+        device_tokens = await cruds_notification.get_firebase_tokens_by_user_ids(
+            user_ids=[loan.borrower_id],
+            db=db,
+        )
+        if (
+            await cruds_notification.get_messages_by_context_and_firebase_tokens(
+                context=f"loan-new-{loan.id}-end-notif",
+                firebase_tokens=device_tokens,
+                db=db,
+            )
+            != []
```

Comment:
If a user has two devices, a message may still be in database for a device, but already delivered for the other. I think you always need to send the notification.